cpp-package/src/board_shim.cpp

```diff
@@ -168,6 +168,23 @@ BrainFlowArray<double, 2> BoardShim::get_board_data ()
     delete[] buf;
     return matrix;
 }
+BrainFlowArray<double, 2> BoardShim::get_board_data (int num_datapoints)
+{
+    int num_samples = get_board_data_count ();
+    num_samples = std::min(num_samples, num_datapoints);
```

Comment:
there is a CI pipeline that checks code style and it will be failed, also there is a tool to format code automatically, take a look at https://brainflow.readthedocs.io/en/stable/BrainFlowDev.html#code-style

Reply:
and let's add a check that this value is bigger than 0(the same for all languages), otherwise, it should throw an exception with INVALID_ARGUMENTS_ERROR code